src/vm-vcpu/src/vm.rs

```diff
@@ -229,7 +231,11 @@ impl<EH: 'static + ExitHandler + Send> KvmVm<EH> {
         let mut vm = Self::create_vm(kvm, vm_config, exit_handler, guest_memory)?;
 
         #[cfg(target_arch = "x86_64")]
-        MpTable::new(vm.config.num_vcpus)?.write(guest_memory)?;
+        {
+            let mp_table = MpTable::new(vm.config.num_vcpus)?;
```

Comment:
I think it makes more sense to send the IRQ as a parameter for the MpTable similarly to how we are setting the vcpu number. This is also what we do on aarch64 when passing the IRQ number to the GIC config.

Reply:
Yes, I was going first for finalising the api for aarch64 of passing down the irq, will fix x86 next.